apps/server/src/shared/domain/entity/system.entity.ts

```diff
@@ -64,6 +66,74 @@ export class OauthConfig {
 	@Property()
 	jwksEndpoint: string;
 }
+
+export class LdapConfig {
+	constructor(ldapConfig: LdapConfig) {
+		this.url = ldapConfig.url;
+	}
+
+	@Property()
+	active?: boolean;
+
+	@Property()
+	federalState?: EntityId;
+
+	@Property()
+	lastSyncAttempt?: Date;
+
+	@Property()
+	lastSuccessfulFullSync?: Date;
+
+	@Property()
+	lastSuccessfulPartialSync?: Date;
+
+	@Property()
+	lastModifyTimestamp?: string;
+
+	@Property()
+	url: string;
+
+	@Property()
+	rootPath?: string;
+
+	@Property()
+	searchUser?: string;
+
+	@Property()
+	searchUserPassword?: string;
+
+	@Property()
+	provider?: string;
+
+	@Property()
+	providerOptions?: {
+		schoolName?: string;
+		userPathAdditions?: string;
+		classPathAdditions?: string;
+		roleType?: string;
+		userAttributeNameMapping?: {
+			givenName?: string;
+			sn?: string;
+			dn?: string;
+			uuid?: string;
+			uid?: string;
+			mail?: string;
+			role?: string;
```

Comment:
If role is complet free or can we define a enum of strings? 
Maybe for some other keys too?

Can you please check this?

Reply:
This can not be an enum. It is the name of the attribute in the object returned by LDAP to determine the role. It is freely editable by the admin.